projects/testing-library__dom/index.js

```diff
@@ -0,0 +1,54 @@
+HTMLDocument.prototype.querySelectorAllWithShadowDOM = function(query) {
+    const isNotCustomElementParent = true;
+    const children = [...(this.children || [])].filter(
+        (child) => isNotCustomElementParent || !!child.assignedSlot
+    );
+    let results = children.filter((el) => el.matches(query));
+    if (this.shadowRoot) {
+        results = results.concat([
+            ...(this.shadowRoot.querySelectorAllWithShadowDOM(query) || []),
+        ]);
+    }
+    children.map((child) => {
+        if (child.querySelectorAllWithShadowDOM) {
+            results = results.concat([
+                ...(child.querySelectorAllWithShadowDOM(query) || []),
+            ]);
+        }
+    });
+    return results;
+};
+
+HTMLElement.prototype.querySelectorAllWithShadowDOM = function(query) {
+    const isNotCustomElementParent = this.tagName.search('-') === -1;
+    const children = [...(this.children || [])].filter(
+        (child) => isNotCustomElementParent || !!child.assignedSlot
+    );
+    let results = children.filter((el) => el.matches(query));
+    if (this.shadowRoot) {
+        results = results.concat([
+            ...(this.shadowRoot.querySelectorAllWithShadowDOM(query) || []),
+        ]);
+    }
+    children.map((child) => {
+        if (child.querySelectorAllWithShadowDOM) {
+            results = results.concat([
+                ...(child.querySelectorAllWithShadowDOM(query) || []),
+            ]);
+        }
+    });
+    return results;
+};
+
+ShadowRoot.prototype.querySelectorAllWithShadowDOM = function(query) {
+    const children = [...(this.children || [])];
+    let results = children.filter((el) => el.matches(query));
+    children.map((child) => {
+        results = results.concat([
+            ...(child.querySelectorAllWithShadowDOM(query) || []),
+        ]);
+    });
```

Comment:
This section appears to be missing a check of the property before invoking the function, e.g.
```suggestion
    children.map((child) => {
        if (child.querySelectorAllWithShadowDOM) {
            results = results.concat([
                ...(child.querySelectorAllWithShadowDOM(query) || []),
            ]);    
        }
    });
```

I've hit this error in my own tests, while adding the check seems to introduce no adverse effects.

Reply:
Definitely makes sense!

You wouldn't happen to have any extra information about the type of `child` that it's been triggering this issue for you on? Want to make sure we're applying the method to all of the required prototypes along with making a check/gate like this for backup safety.

Reply:
The node that triggers this is an instance of [`SVGSVGElement`](https://developer.mozilla.org/en-US/docs/Web/API/SVGSVGElement).